cpp/oneapi/dal/backend/primitives/blas/syrk_dpc.cpp

```diff
@@ -0,0 +1,91 @@
+/*******************************************************************************
+* Copyright 2022 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include "oneapi/dal/detail/profiler.hpp"
+
+#include "oneapi/dal/backend/primitives/blas/syrk.hpp"
+#include "oneapi/dal/backend/primitives/blas/misc.hpp"
+
+#include <mkl_dal_sycl.hpp>
+
+namespace oneapi::dal::backend::primitives {
+
+template <typename Float>
+static sycl::event syrk_wrapper(sycl::queue& queue,
+                                mkl::uplo uplo,
+                                mkl::transpose trans,
+                                std::int64_t n,
+                                std::int64_t k,
+                                Float alpha,
+                                const Float* a,
+                                std::int64_t lda,
+                                Float beta,
+                                Float* c,
+                                std::int64_t ldc,
+                                const event_vector& deps) {
+#ifdef ONEDAL_ENABLE_ASSERT
+    const bool is_trans = (trans == mkl::transpose::trans);
+    ONEDAL_ASSERT(ldc >= n);
+    ONEDAL_ASSERT(is_trans || lda >= n);
+    ONEDAL_ASSERT(!is_trans || lda >= k);
+#endif
```

Comment:
`#ifdef ONEDAL_ENABLE_ASSERT` is inside `ONEDAL_ASSERT` define already

Reply:
Changed to use of ```[[maybe_unused]]`` instead